stdlib/lists.ncl

```diff
@@ -0,0 +1,37 @@
+{
+  lists = {
+    concat = Promise(List -> List -> List, fun l1 l2 => l1 @ l2);
+
+    foldl = Promise(forall a. (a -> Dyn -> a) -> a -> List -> a,
```

Comment:
I'm rather surprised that we don't have a type `List b`. But I suppose this can be solved later.

Reply:
True. Saved in #171.